pkg/pipelines/environment.go

```diff
@@ -0,0 +1,84 @@
+package pipelines
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/openshift/odo/pkg/pipelines/meta"
+	"github.com/openshift/odo/pkg/pipelines/roles"
+)
+
+const (
+	envNamespace   = "namespace.yaml"
+	envRoleBinding = "rolebinding.yaml"
+)
+
+// EnvParameters encapsulates parameters for add env command
+type EnvParameters struct {
+	EnvName    string
+	Output     string
+	Prefix     string
+	GitOpsRepo string
+}
+
+// Env will bootstrap a new environment directory
+func Env(o *EnvParameters) error {
+
+	gitopsName := getGitopsRepoName(o.GitOpsRepo)
+	gitopsPath := filepath.Join(o.Output, gitopsName)
+
+	envName := addPrefix(o.Prefix, o.EnvName)
+	envPath := getEnvPath(gitopsPath, o.EnvName, o.Prefix)
+
+	// check if the gitops dir exists
+	exists, _ := isExisting(gitopsPath)
+	if !exists {
+		return fmt.Errorf("%s doesn't exist at %s", gitopsName, o.Output)
+	}
+
+	// check if the environment dir already exists
+	exists, _ = isExisting(envPath)
+	if exists {
+		return fmt.Errorf("%s already exists", envName)
+	}
+
+	err := addKustomize("resources", []string{envNamespace, envRoleBinding}, filepath.Join(envPath, "base", kustomize))
+	if err != nil {
+		return err
+	}
+
+	err = addKustomize("bases", []string{"../base"}, filepath.Join(envPath, "overlays", kustomize))
+	if err != nil {
+		return err
+	}
+
+	if err = addEnvResources(o.Prefix, envPath, envName); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func addEnvResources(prefix, envPath, envName string) error {
+
+	namespaces := namespaceNames(prefix)
```

Comment:
We need to make sure the user passes a consistent prefix as the "init" sub-commend.   Otherwise, these namespaces won't match.

Reply:
If the "odo pipelines init" ultimately generates manifests, we can read manifests for the "prefix" when we exec "pipelines env add" and doing what you have here is fine for the time being.

Reply:
A similar issue with --output and --gitops-repo options, the only reason this "environment" command need them is because they are used to locate the root directory.   When we have the manifest in place, these options will need to be revisited.

Reply:
Can I resolve this conversation as discussed, as we would revisiting this during issue 85 https://issues.redhat.com/browse/GITOPS-85 ?